static/char_counter/js/char_counter.js

```diff
@@ -0,0 +1,52 @@
+let LineLengthSettingSingleton = 0;
+
+document.addEventListener('DOMContentLoaded', () => {
+	LineLengthSettingSingleton = getLineLength();
+
+	const inputs = document.querySelectorAll('.char-counter-text-input')
+
+	inputs.forEach((input) => {
+		input.addEventListener('input', e => updateCharCountLabel(e.target));
+	});
+
+	document.querySelector('#id_line_length').addEventListener('change', (e) => {
+		LineLengthSettingSingleton = e.target.value;
+
+		inputs.forEach(input => {
+			updateCharCountLabel(input)
+		})
+	});
+})
+
+function updateCharCountLabel(input) {
+	const label = findLabel(input);
+	const valueLength = input.value.length;
+
+	if (valueLength > LineLengthSettingSingleton) {
+		addClass(label, 'long');
+	} else {
+		removeClass(label, 'long');
+	}
+
+	label.textContent = `${valueLength} chars`;
+}
+
+function findLabel(eventTarget) {
+	return eventTarget.parentNode.querySelector('.char-counter-text-input + .counter');
```

Comment:
The `+` means "next sibling" – so it might be a little more specific than we need. I never know how specific to make my DOM selectors – do I be as specific as possible to try and avoid unintended matches? Or do I be as general as possible to try and avoid brittle selectors. Thoughts?

Reply:
ew.

Reply:
so, `'.char-counter-text-input` matches one thing and then `+ .counter`' matches the next sibling in the DOM that has `.counter` class?

Reply:
yeah so only `.counter`s who are the next sibling of `.char-counter-text-input`